Makefile

```diff
@@ -0,0 +1,103 @@
+# Makefile
+
+WARNINGS=RUSTDOCFLAGS="-D warnings"
+DEBUG_ASSERTIONS=RUSTFLAGS="-C debug-assertions"
+
+# -- linting --------------------------------------------------------------------------------------
+
+# Runs Clippy with configs
+.PHONY: clippy
+clippy:
+	cargo clippy --workspace --all-targets --all-features -- -D warnings
+
+# Runs Fix with configs
+.PHONY: fix
+fix:
+	cargo +nightly fix --allow-staged --allow-dirty --all-targets --all-features
+
+# Runs Format using nightly toolchain
+.PHONY: format
+format:
+	cargo +nightly fmt --all
+
+# Runs Format using nightly toolchain but only in check mode
+.PHONY: format-check
+format-check:
+	cargo +nightly fmt --all --check
+
+# Runs all linting tasks at once (Clippy, fixing, formatting)
+.PHONY: lint
+lint:
+	$(MAKE) format
+	$(MAKE) clippy
+	$(MAKE) fix
```

Comment:
Does the sequencing matter here? For example, should it be `format`, `fix`, and then `clippy`?

Reply:
> I don't see why it would matter, I changed it to as requested format fix clippy

If the order matters, then it should be defined with additional dependencies, e.g.:

```
format: clippy
fix: clippy
```

Because otherwise these rules are executed in parallel, here is an example:

```Makefile
default: a b c

a:
	python3 -c 'import time; time.sleep(1); print("a")'
b:
	python3 -c 'import time; time.sleep(1); print("b")'
c:
	python3 -c 'import time; time.sleep(1); print("c")'
```

Running the `Makefile` above results in:

```
> make -j test1
python3 -c 'import time; time.sleep(1); print("a")'
python3 -c 'import time; time.sleep(1); print("b")'
python3 -c 'import time; time.sleep(1); print("c")'
b
a
c
```